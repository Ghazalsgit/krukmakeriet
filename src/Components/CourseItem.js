import React from 'react'
import styles from "./CourseItem.module.css";
import { useContext } from 'react'
import { AppContext } from '../App'


function CourseItem({courseData}) {

  const {setCourseBasket} = useContext(AppContext) 

  function addToBasket(courseData) {
    setCourseBasket(prevBasket => [...prevBasket, courseData])
    console.log("Added to basket " + courseData.name)
  }
  

  return (
    <article className={styles.courseCardWrapper}>
      <div className={styles.imgWrapper}>
        <img className={styles.courseImage} src={courseData.img} alt="" />
      </div>
      
      <div className={styles.textWrapper}>
        <div className={styles.infoWrapper}>
          <h3 className={styles.name}>{courseData.name}</h3>
          <p className={styles.date}>[ikon]{courseData.details}</p> 
          <p className={styles.info}>{courseData.info}</p>
        </div>
        
        <div className={styles.iconWrapper}>
          <p className={styles.price}>[ikon] {courseData.price}:-</p>
          <p className={styles.length}>[ikon] {courseData.duration}</p>
          <p className={styles.spots}>[ikon] [10/{courseData.spots}]</p>
        </div>
        <button className={styles.button} onClick={() => addToBasket(courseData)}>Boka</button>
      </div>
    
    </article>
  )
}

export default CourseItem