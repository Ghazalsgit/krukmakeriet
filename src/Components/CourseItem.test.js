import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";
import CourseItem from "./CourseItem";
import { AllContextProvider } from "../context/AllContext";
import { BrowserRouter } from "react-router-dom";
import renderer from "react-test-renderer";

const MockCourseItem = () => {
  return (
    <BrowserRouter>
      <AllContextProvider>
        <CourseItem courseData={course} />
      </AllContextProvider>
    </BrowserRouter>
  );
};
const course = { name: "abc", img: "https" };

describe("CourseItem", () => {
  test("check if props exist as should", async () => {
    render(<MockCourseItem />);
    const routeForCourseItem = screen.getByTestId("courseItem");
    expect(routeForCourseItem).toBeInTheDocument();
  });

  test("Matches DOM Snapshot", () => {
    const domTree = renderer
      .create(
        <BrowserRouter>
          <AllContextProvider>
            <CourseItem courseData={course} />
          </AllContextProvider>
        </BrowserRouter>
      )
      .toJSON();
    expect(domTree).toMatchSnapshot();
  });
});
