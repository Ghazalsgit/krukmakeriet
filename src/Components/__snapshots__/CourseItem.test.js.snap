// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CourseItem Matches DOM Snapshot 1`] = `
<article
  className="courseCardWrapper"
  data-testid="courseItem"
>
  <div
    className="imgWrapper"
  >
    <img
      alt=""
      className="courseImage"
      src="https"
    />
  </div>
  <div
    className="textWrapper"
  >
    <div
      className="infoWrapper"
    >
      <h3
        className="name"
      >
        abc
      </h3>
      <p
        className="date"
      >
        [ikon]
      </p>
      <p
        className="info"
      />
    </div>
    <div
      className="iconWrapper"
    >
      <p
        className="price"
      >
        [ikon] 
        :-
      </p>
      <p
        className="length"
      >
        [ikon] 
      </p>
      <p
        className="spots"
      >
        [ikon] [10/
        ]
      </p>
    </div>
    <button
      className="button"
      onClick={[Function]}
    >
      Boka
    </button>
  </div>
</article>
`;
